explore/src/main/scala/explore/observationtree/ObsList.scala

```diff
@@ -119,12 +123,53 @@ object ObsList:
       .useMemoBy((props, _, _, _) => (props.observations, props.groups))((_, _, _, _) =>
         ObsNode.fromList
       )
+      .useEffectWithDepsBy((props, _, _, _, _) => (props.focusedObs, props.groups))(
+        (props, _, _, _, _) =>
+          case (None, _)             => Callback.empty
+          case (Some(obsId), groups) =>
+            @tailrec
+            def findParentGroups(
+              groupElementId: Either[Observation.Id, Group.Id],
+              acc:            Set[Group.Id]
+            ): Set[Group.Id] = {
+              val parentGroup = groups.find(
+                GroupElement.grouping
+                  .exist(_.elements.exists(_.bimap(_.id, _.id) === groupElementId))
+              )
+
+              parentGroup match
+                case None                                                => acc
+                case Some(GroupElement(Left(_), _))                      => acc
+                // We've found the 'root' group, so we're done
+                case Some(GroupElement(Right(grouping), None))           => acc + grouping.id
+                case Some(GroupElement(Right(grouping), Some(parentId))) =>
+                  findParentGroups(parentId.asRight, acc ++ Set(parentId, grouping.id))
+            }
+
+            val groupsToAddFocus =
+              findParentGroups(obsId.asLeft, Set.empty)
+
+            Callback.when(groupsToAddFocus.nonEmpty)(
+              props.expandedGroups.mod(_ ++ groupsToAddFocus)
+            )
+      )
       .render { (props, ctx, _, adding, treeNodes) =>
 
         import ctx.given
 
         val observations = props.observations.toList
 
+        // TODO: remove after https://github.com/gemini-hlsw/lucuma-react/pull/465 is merged and released
+        val expandedGroups =
+          props.expandedGroups.zoom(_.map(k => Tree.Id(k.toString) -> true).toMap)(f =>
```

Comment:
We'll still need to convert from Set[Group.Id] to Set[Tree.Id] somewhere, right? Maybe we could memoize that or something?

Reply:
We could also keep a `Set[Tree.Id]`, but that feels slightly less clean (since it is always for a group)

Reply:
Agreed